hydromt_fiat/fiat.py

```diff
@@ -149,100 +305,348 @@ def setup_exposure_vector(
+            for all the elements a single value can be provided as integer, by default
+            None.
+        var : Union[str, list[str], None], optional
+            The name of the variable to be selected in case a netCDF file is provided
+            as input, by default None.
+        chunks : Union[int, str, list[int]], optional
+            The chuck region per map. In case a single no data applies for all the
+            elements a single value can be provided as integer. If "auto"is provided
+            the auto setting will be provided by default "auto"
+        hazard_type : str, optional
+            Type of hazard to be studied, by default "flooding"
+        risk_output : bool, optional
+            The parameter that defines if a risk analysis is required, by default False
+        """
+        # check parameters types and size, and existance of provided files of maps
+
+        params = check_parameters_type(map_fn, map_type, rp, crs, nodata, var, chunks)
+        check_parameters_size(params)
+        check_files(params, self.root)
+
+        rp_list = []
+        map_name_lst = []
+
+        # retrieve maps information from parameters and datacatalog
+        # load maps in memory and check them and save the with st_map function
+        for idx, da_map_fn in enumerate(params["map_fn_lst"]):
+            da_map_fn, da_name, da_type = read_floodmaps(params, da_map_fn, idx)
+
+            # load flood maps to memory
+            # da = load_floodmaps(self.data_catalog, self.region,da_map_fn,da_name,name_catalog)
+            # reading from path
+            if isinstance(da_map_fn, Path):
+                if da_map_fn.stem == "sfincs_map":
+                    sfincs_root = os.path.dirname(da_map_fn)
+                    sfincs_model = SfincsModel(
+                        sfincs_root, mode="r", logger=self.logger
+                    )
+                    sfincs_model.read_results()
+                    # save sfincs map as GeoTIFF
+                    # result_list = list(sfincs_model.results.keys())
+                    # sfincs_model.write_raster("results.zsmax", compress="LZW")
+                    da = sfincs_model.results["zsmax"]
+                    # da = da.squeeze('timemax').drop('timemax')
+                    da = da.isel(timemax=0).drop("timemax")
+
+                    # Convert to units of the exposure data if required
+                    if self.exposure.unit != da.units:
+                        da = da * unit_conversion_factor
+
+                else:
+                    if not self.region.empty:
+                        # da = self.data_catalog.get_rasterdataset(
+                        #     da_map_fn, geom=self.region
+                        # )
+                        da = self.data_catalog.get_rasterdataset(da_map_fn)
+                    else:
+                        da = self.data_catalog.get_rasterdataset(da_map_fn)
```

Comment:
@frederique-hub, we can use any of the options below. I decided to keep the if statement because self.region was not always created. We have to ensure that self.region is always created to always go for the 2) option and remove the if statement.

            1) da = self.data_catalog.get_rasterdataset(da_map_fn)
            2) da = self.data_catalog.get_rasterdataset(da_map_fn, geom=self.region)